### PR TITLE
P2579R0 Mitigation strategies for P2036 "Changing scope for lambda trailing-return-type"

### LINKED ISSUE #5586: [2022-07 CWG Motion 21] P2579R0 Mitigation strategies for P2036 ”Changing scope for lambda trailing-return-type”
[P2579R0](https://wg21.link/p2579r0)

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #5586
Fixes cplusplus/papers#1274

### CODE REVIEW

File: source/expressions.tex [hunk 1]
```diff
@@ -1500,15 +1501,14 @@
     return y2;
   };
 
-  [=]<decltype(x) P>{};         // error: \tcode{x} refers to local entity but precedes the
-                                // lambda's function parameter scope
-  [=](decltype((x)) y){};       // error: \tcode{x} refers to local entity but is in the lambda's
-                                // \grammarterm{parameter-declaration-clause}
+  [=](decltype((x)) y){
+    decltype((x)) z = x;        // Ok, \tcode{y} has type \tcode{float\&}, \tcode{z} has type \tcode{float const\&}
```
Comment (JohelEGP) [inline source/expressions.tex:1505]:
```suggestion
    decltype((x)) z = x;        // OK, \tcode{y} has type \tcode{float\&}, \tcode{z} has type \tcode{float const\&}
```

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/expressions.tex [hunk 2]
```diff
@@ -2242,22 +2242,26 @@
 is said to be \defn{explicitly captured}.
 
 \pnum
-If an \grammarterm{identifier} in a \grammarterm{simple-capture} appears
```
Comment (tkoeppe) [inline source/expressions.tex:2245]: @jensmaurer: The paper says,

> The changes to this paragraph have not been explicitely discussed
by EWG, and may be better treated as a core issue or an NB comment.

Is CWG aware of this and has the straw poll motion been taken with this information well-known?

Comment (jensmaurer) [reply source/expressions.tex:2245]: CWG has seen the wording as we're seeing it now, so was aware of this.

(CWG chair) This feels like an obvious consistency fix to me that doesn't need EWG input.

Comment (tkoeppe) [reply source/expressions.tex:2245]: That's great, thanks!

**Comment by Mick235711** [comment]
Also fixes cplusplus/papers#1227. (CWG 2569 can be marked as WP too)

**Pull request merged**